src/vignette/util/image_response.clj

```diff
@@ -38,13 +38,14 @@
 
 (defn create-image-response
   ([image image-map]
-   (-> (response (->response-object image))
-       (header "Content-Type" (content-type image))
-       (header "Content-Length" (content-length image))
-       (header "ETag" (etag image))
-       (header "X-Thumbnailer" "Vignette")
-       (add-content-disposition-header image-map)
-       (add-surrogate-header image-map)))
+
+   (as-> (response (->response-object image)) resp
+         (if-let [ct (content-type image)] (header resp "Content-Type" ct) resp)
+         (if-let [cl (content-length image)] (header resp "Content-Length" cl) resp)
+         (if-let [etag (etag image)] (header resp "ETag" etag) resp)
```

Comment:
Those ifs are actually from time when SAS was under active development and didn't provide some of the headers that resulted in errors.
I left them because if such situation were to ever happen again this would allow vignette to still show images.